### ISSUE TITLE
[macOS] java.nio.file.NoSuchFileException when launching child Eclipse

### ISSUE THREAD

**Issue by Phillipus** [eclipse-pde/eclipse.pde#458]
Not sure if PDE is the right place to report this, please advise if wrong.

- macOS 12.6.2 and Ventura 13.2
- Eclipse 2022-12 (4.26)
- Java 17.0.5

1. Create a new run configuration to run product `org.eclipse.sdk.ide` with the `org.eclipse.sdk` Feature. All other settings are default.
2. Run/Debug the configuration

```
java.nio.file.NoSuchFileException: file:/<workspace>/.metadata/.plugins/org.eclipse.pde.core/.bundle_pool/Contents/Info.plist
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:380)
	at java.base/java.nio.file.Files.newByteChannel(Files.java:432)
	at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:422)
	at java.base/java.nio.file.Files.newInputStream(Files.java:160)
	at java.base/java.nio.file.Files.newBufferedReader(Files.java:2922)
	at java.base/java.nio.file.Files.newBufferedReader(Files.java:2955)
	at org.eclipse.urischeme.internal.registration.FileProvider.newReader(FileProvider.java:45)
	at org.eclipse.urischeme.internal.registration.RegistrationMacOsX.getPlistFileWriter(RegistrationMacOsX.java:122)
	at org.eclipse.urischeme.internal.registration.RegistrationMacOsX.changePlistFile(RegistrationMacOsX.java:139)
	at org.eclipse.urischeme.internal.registration.RegistrationMacOsX.handleSchemes(RegistrationMacOsX.java:55)
	at org.eclipse.urischeme.AutoRegisterSchemeHandlersJob.run(AutoRegisterSchemeHandlersJob.java:94)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by gireeshpunathil** [comment]
@Phillipus - looks to me like the run was issued in a stale workspace that earlier used a bundle pool but not anymore. could you try launching the child in a fresh workspace?

**Comment by Phillipus** [comment]
@gireeshpunathil Thanks for the hint. I deleted my `<workspace>/.metadata/.plugins/org.eclipse.pde.core` folder and cleaned everything and the problem has gone.

**Comment by travkin79** [comment]
I had a very similar case on my Mac (MacOS Sonoma 14.4.1, Eclipse 2024-03, JDK 18). In addition to the missing `Contents/Info.plist` file, I observed strange debugger behavior like not finding classes when running an RCP application in debug mode (but I had no problems with the same application and launch config when run in non-debug mode).

First, I thought, I would have to create a new runtime workspace for the application under test, but it turns out, I had to re-create my parent Eclipse IDE's workspace (my main workspace), since the `Info.plist` that was not found in the child Eclipse IDE came from the parent Eclipse IDE's workspace.

I'm not sure if the problem arose after updating Eclipse to 2024-03 and migrating the workspace.
